pkg/crc/preflight/preflight_checks_linux.go

```diff
@@ -472,6 +515,26 @@ func fixCrcDnsmasqConfigFile() error {
 	return nil
 }
 
+func removeCrcDnsmasqConfigFile() error {
+	if _, err := os.Stat(crcDnsmasqConfigPath); !os.IsNotExist(err) {
+		logging.Debug("Removing dnsmasq configuration")
+		err := crcos.RemoveFileAsRoot(
+			fmt.Sprintf("removing dnsmasq configuration in %s", crcDnsmasqConfigPath),
+			crcDnsmasqConfigPath,
+		)
+		if err != nil {
+			return fmt.Errorf("Failed to remove dnsmasq config file: %s: %v", crcDnsmasqConfigPath, err)
+		}
+
+		logging.Debug("Reloading NetworkManager")
+		sd := systemd.NewHostSystemdCommander()
+		if _, err := sd.Reload("NetworkManager"); err != nil {
+			return fmt.Errorf("Failed to restart NetworkManager: %v", err)
```

Comment:
This could fail when `skip-check-network-manager-installed` is set.

Reply:
@cfergeau we are adding the dnsmasqConfig file to the `/etc/NetworkManager` location so even setup fails if Network manager is not installed and `skip-check-network-manager-installed` is set to true.

Reply:
Ah it  fails because the `/etc/NetworkManager` directory does not exist. This does not seem like something intentional though, just luck ;)

Reply:
@cfergeau not luck, right now our dnsmasq configuration depends on NM so all those config only happen when NM is installed.

Reply:
I don't see anything in the code making sure `fixCrcNetworkManagerConfig` and `fixCrcDnsmasqConfigFile` are only run if NetworkManager is installed and running (in other word, they don't check if `checkNetworkManagerIsRunning` and `checkNetworkManagerInstalled` return nil). They indeed won't work if NetworkManager is not installed, but only because they will try to write in directories which don't exist. To me this does not need like something intentional, just a corner-case/side effect which happens to do the right thing in our case.

I'd prefer if we did not add more code relying on that assumption in order to work properly when it seems relatively easy to avoid it (ignoring the error seems like it's going to achieve this?)